### PR TITLE
fix some error for windows

### PR DESCRIPTION
**Pull request by falseen**

### CODE REVIEW

**Comment by guedou** [comment]
Thanks for your PR. Could you please provide use some hints about your issue ?

**Comment by gpotter2** [comment]
Hi, 
What you,be done is interesting. But instead of quick fixing stuff, you could have submitted an issue request, or a full working pull request. As you said:
> This is a temporary solution.

Apart from that I don't see why all of those new imports are useful..

**Comment by falseen** [comment]
@gpotter2
Oh, sorroy. My English is not very good.
I don't know the specific reason for the error, I think it is a iface's name was too long.

My work environment: win7 32bit python27

File: scapy/arch/windows/__init__.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def _exec_query_ps(cmd, fields):
     for line in ps.stdout:
         if not line.strip(): #skip empty lines
             continue
-        l.append(line.split(':', 1)[1].strip())
+        l.append(line.split(':', 1)[-1].strip())
```
Comment (gpotter2) [inline scapy/arch/windows/__init__.py]: This is a quick fix... Not that good

File: scapy/arch/windows/compatibility.py [hunk 2]
```diff
@@ -9,6 +9,10 @@
 
 from scapy.arch.consts import LOOPBACK_NAME
 from scapy.config import conf,ConfClass
+from scapy.error import warning, Scapy_Exception, log_runtime
+from scapy.data import *
+from scapy.arch import pcapdnet
+from scapy.arch.pcapdnet import *
```
Comment (gpotter2) [inline scapy/arch/windows/compatibility.py]: I do not really understand why all of this is useful, wasn't it already working ?

File: scapy/arch/windows/compatibility.py [hunk 3]
```diff
@@ -9,6 +9,10 @@
...
```
Comment (gpotter2) [inline scapy/arch/windows/compatibility.py]: Some of those are relevent, some using "*" are not

**Comment by gpotter2** [comment]
So to conclude, here are the required missing imports:
```
from scapy.base_classes import Gen, SetGen
import scapy.plist as plist
from scapy.utils import PcapReader
from scapy.data import MTU, ETH_P_ARP
```

File: scapy/arch/windows/compatibility.py (hunk 3 above)
Comment (gpotter2) [inline scapy/arch/windows/compatibility.py]: Yes, as some imports are missing. See below for a better fix

File: scapy/arch/windows/__init__.py (hunk 1 above)
Comment (falseen) [inline scapy/arch/windows/__init__.py]: The 'line.split(':', 1)[1]' and 'line.split(':', 1)[-1]'  is no difference, and the latter can avoid mistakes.

**Comment by falseen** [comment]
@gpotter2
I found another problem, when running sniffers (), it will throw the following error:

```python
>>> from scapy.all import *
>>> sniff(count = 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "scapy\arch\windows\compatibility.py", line 219, in sniff
    return plist.PacketList(lst,"Sniffed")
NameError: global name 'plist' is not defined
```
But the plist has been imported in the front.

**Comment by gpotter2** [comment]
Oh You should add 
``WINDOWS = True``
too at the beggining of the comptability.py file

**Comment by gpotter2** [comment]
There are too many changes in a single PR i would say... Some are really useful, some do not seems so...
I've created a PR with only the import fix as it (in my opinion) is the most important part of it, and can quickly be merged.

**Comment by guedou** [comment]
Shall we close this PR ?

**Comment by gpotter2** [comment]
Yes, any other things should be added in separate PR

**Pull request closed without merging**